docs/writing-tests/testharness-tutorial.md

```diff
@@ -0,0 +1,390 @@
+# testharness.js tutorial
+
+<!--
+Note to maintainers:
+
+This tutorial is designed to be an authentic depiction of the WPT contribution
+experience. It is not intended to be comprehensive; its scope is intentionally
+limited in order to demonstrate authoring a complete test without overwhelming
+the reader with features. This should be updated as typical WPT usage patterns
+change, but please weigh extensions against the demotivating effect that a
+lengthy guide can have on new contributors.
+-->
+
+Let's say you've discovered that WPT doesn't have any tests for how [the Fetch
+API](https://fetch.spec.whatwg.org/) sets cookies from an HTTP response. This
+tutorial will guide you through the process of writing and submitting a test
+for exactly that. Although it includes some very brief instructions on using
+git, you can find more guidance in [the tutorial for git and
+GitHub](../appendix/github-intro).
+
+WPT's testharness.js is a framework designed to help people write tests for the
+web platform's JavaScript APIs. [The testharness.js reference
+page](testharness) describes the framework in the abstract, but for the
+purposes of this guide, we'll take it for granted that testharness.js is the
+best way to test the behavior of `fetch`.
+
+```eval_rst
+.. contents::
+   :local:
+```
+
+## Setting up your workspace
+
+To make sure you have the latest code, first type the following into a terminal
```

Comment:
Do they need to clone and fork it first?

Reply:
They do. The second half of this instruction reads, "located in the root of the
WPT git repository." I worded it this way to limit the amount of duplication
between this tutorial, the reftest tutorial, and the guide to using git and
GitHub. Readers have already been advised to use the git guide as needed, so I
think this is a fair way to abbreviate the content without concealing necessary
information.

Reply:
OK, works for me.